lib/Sema/TypeCheckType.cpp

```diff
@@ -3453,6 +3472,11 @@ void TypeChecker::checkUnsupportedProtocolType(Decl *decl) {
       decl->getDeclContext()->isModuleScopeContext())
     return;
 
+  // Global variable declarations are okay, because they'll be caught in the stmt checker.
```

Comment:
@slavapestov Yes, just verified that. In -parse-as-library mode there isn't a top-level statement, but there is a top-level pattern decl instead, and the checking on that pattern decl catches the unsupported protocol type in the same source code location. I've updated the comment here to be more accurate.